engine/runtime/src/main/java/org/enso/interpreter/runtime/builtin/Error.java

```diff
@@ -126,6 +148,24 @@ public Atom makeNoSuchMethod(Object target, UnresolvedSymbol symbol) {
     return noSuchMethod.newInstance(target, symbol);
   }
 
+
+  /** Verifies payload whether it represents {@link #makeNoSuchMethod(Object, UnresolvedSymbol)} atom.
+   * @param payload object to check
+   * @return {@code null} if the {@code payload} isn't recognized name of the unresolved symbol otherwise
+  */
+  public String findNoSuchMethod(Object payload) {
+    if (payload instanceof Atom atom && atom.getConstructor() == noSuchMethod.getUniqueConstructor()) {
```

Comment:
> now we have essentially 2 errors to indicate the same situation

The goal of this `findNoSuchMethod` equilibristic is to avoid _two errors situation_. Internally we continue to represent these errors as `PanicException`.

>  ...replace No_Such_Method with UnknownIdentifierException only in some places?

Yes, exactly. Whenever the `PanicException` crosses the boundary and propagates outside of Enso, it may get converted, so other languages and systems like the one handling `Fooable` interface can recognize `UnknownIdentifierException` and handle it in better way.

When they saw just `PanicException` they treated it as _runtime error_ in user code, not as _unknown identifier_ - e.g. problem finding/invoking the `toString()` completely.

Reply:
> The goal of this `findNoSuchMethod` equilibristic is to avoid _two errors situation_. Internally we continue to represent these errors as `PanicException`.

So does it mean that in Enso, I will _always_ see the Panic `No_Such_Method` and never the `UnknownIdentifierException` which is only meant for other languages?